static/js/settings_org.js

```diff
@@ -543,6 +543,41 @@ export let default_code_language_widget = null;
 export let notifications_stream_widget = null;
 export let signup_notifications_stream_widget = null;
 
+export function get_widget_for_dropdown_list_settings(property_name) {
+    switch (property_name) {
+        case "realm_notifications_stream_id":
+            return notifications_stream_widget;
+        case "realm_signup_notifications_stream_id":
+            return signup_notifications_stream_widget;
+        case "realm_default_code_block_language":
+            return default_code_language_widget;
+        default:
+            blueslip.error("No dropdown list widget for property: " + property_name);
+            return null;
+    }
+}
+
+export function set_dropdown_list_widget_setting_value(property_name, value) {
+    const widget = get_widget_for_dropdown_list_settings(property_name);
+    widget.render(value);
+}
+
+export function get_dropdown_list_widget_setting_value($input_elem, for_api_data = true) {
+    const widget_name = extract_property_name($input_elem);
+    const setting_widget = get_widget_for_dropdown_list_settings(widget_name);
+
+    const setting_value_type = $input_elem.attr("data-setting-value-type");
+    if (setting_value_type === "number") {
+        return Number.parseInt(setting_widget.value(), 10);
+    }
+
+    const setting_value = setting_widget.value();
+    if (setting_value.length === 0 && !for_api_data) {
```

Comment:
Maybe add a comment before this block explaining reason we have this `null` quirk for `!for_api_data` to make this function more readable.